src/hotspot/share/opto/ifnode.cpp

```diff
@@ -1015,63 +1015,56 @@ bool IfNode::fold_compares_helper(ProjNode* proj, ProjNode* success, ProjNode* f
       const TypeInt* type2 = filtered_int_type(igvn, n, fail);
       if (type2 != nullptr) {
         failtype = failtype->join(type2)->is_int();
-        if (failtype->_lo > failtype->_hi) {
+        if (failtype->empty()) {
           // previous if determines the result of this if so
           // replace Bool with constant
           igvn->replace_input_of(this, 1, igvn->intcon(success->_con));
           return true;
         }
       }
     }
-    lo = nullptr;
-    hi = nullptr;
+    return false;
+  }
+
+  Node* hook = new Node(lo); // Add a use to lo to prevent him from dying
```

Comment:
Would you mind adding a sanity assert, that both `lo` and `hi` are not `nullptr`?